crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -108,6 +128,26 @@ pub struct Polyline2d<const N: usize> {
 }
 impl<const N: usize> Primitive2d for Polyline2d<N> {}
 
+impl<const N: usize> FromIterator<Vec2> for Polyline2d<N> {
+    fn from_iter<I: IntoIterator<Item = Vec2>>(iter: I) -> Self {
+        let mut vertices: [Vec2; N] = [Vec2::ZERO; N];
+
+        let iter = iter.into_iter().take(N);
```

Comment:
You should be able to collapse this into the line below :)

Reply:
Ok I turned it into
```rs
for (index, i) in iter.into_iter().take(N).enumerate() {
```